src/main/scala/com/gradle/internal/Utils.scala

```diff
@@ -120,26 +93,26 @@ object Utils {
     val f = Future(blocking(p.exitValue())) // wrap in Future
     try {
       if (Await.result(f, duration.Duration(10, "sec")) != 0) None
-      else Some(trimAtEnd(output.result()))
+      else Some(trimAtEnd(output.result())).filter(_.nonEmpty)
     } catch {
       case _: TimeoutException =>
         p.destroy()
         p.exitValue()
         None
-      case _@(_: IOException | _: InterruptedException) => None
+      case _ @(_: IOException | _: InterruptedException) => None
     }
   }
 
   private[gradle] def readPropertiesFile(name: String) = {
-      val input = new FileInputStream(name)
-      try {
-          val properties = new Properties
-          properties.load(input)
-          properties
-      } catch {
-          case e: IOException =>
-              throw new RuntimeException(e)
-      } finally if (input != null) input.close()
+    val input = new FileInputStream(name)
```

Comment:
is there a convenient `try with resources` that scala provides for this?

Reply:
Not in Scala 2.12 (which we use), unfortunately.